app/main.py

```diff
@@ -1,6 +1,17 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    @wraps(func)
+    def inner(*args: list) -> list:
+        if args not in cached_results:
```

Comment:
Using `args` directly as a key in `cached_results` can lead to errors if `args` contains unhashable types like lists. Consider converting `args` to a hashable type, such as a tuple of tuples, if it contains mutable elements.